File: src/main/java/com/loltft/rudefriend/config/JwtProperties.java
package com.loltft.rudefriend.config;

import lombok.Getter;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
@Getter
public class JwtProperties {

  @Value("${jwt.secret}")
  private String secret;

  @Value("${jwt.hash-secret}")
  private String hashSecret;

  private final long expiration = 86400000; // 24H
  private final long refreshExpiration = 86400000 * 2; // 48H

  private final String accessTokenSubject = "access_token";
  private final String refreshTokenSubject = "refresh_token";

  private final String claimKey = "memberId";

  private final String accessHeaderName = "AccessToken";
  private final String refreshHeaderName = "RefreshToken";
}

File: src/main/java/com/loltft/rudefriend/config/JwtProperties.kt
package com.loltft.rudefriend.config

import org.springframework.boot.context.properties.ConfigurationProperties


@ConfigurationProperties(prefix = "jwt")
data class JwtProperties(
    val secret: String,
    val hashSecret: String,
    val expiration: Long = 86_400_000, // 24H
    val refreshExpiration: Long = 604_800_000, // 7Day
    val accessTokenSubject: String = "access_token",
    val refreshTokenSubject: String = "refresh_token",
    val claimKey: String = "memberId",
    val accessHeaderName: String = "AccessToken",
    val refreshCookieKey: String = "RefreshToken"
)